### PR TITLE
Stop using deprecated ControllerManagerConfig in controller-runtime pkg

### PR DESCRIPTION
**Pull request by ELENAGER**
Improvements

### CODE REVIEW

**Review by ShyamsundarR — COMMENTED** [review]
Initial glance at the contents seem fine.

Ramen config has a [webhook](https://github.com/RamenDR/ramen/blob/1966369dc12b1797d3a7dffd5a81743ba4689173/config/hub/manager/ramen_manager_config.yaml#L7-L8) section that potentially should be removed? @rakeshgm

**Comment by rakeshgm** [comment]
> Initial glance at the contents seem fine.
> 
> Ramen config has a [webhook](https://github.com/RamenDR/ramen/blob/1966369dc12b1797d3a7dffd5a81743ba4689173/config/hub/manager/ramen_manager_config.yaml#L7-L8) section that potentially should be removed? @rakeshgm

yes @ShyamsundarR , we can remove the webhook section from ramenConfig.

**Review by raghavendra-talur — APPROVED** [review]

File: internal/controller/drcluster_mmode_test.go [hunk 1]
```diff
@@ -98,14 +97,12 @@ var _ = Describe("DRClusterMModeTests", Ordered, func() {
...
 			},
-			ControllerManagerConfigurationSpec: controller_runtime_config.ControllerManagerConfigurationSpec{
-				LeaderElection: &config.LeaderElectionConfiguration{
-					LeaderElect:  new(bool),
-					ResourceName: ramencontrollers.HubLeaderElectionResourceName,
-				},
-				Metrics: controller_runtime_config.ControllerMetrics{
-					BindAddress: "0", // Disable metrics
-				},
+			LeaderElection: &config.LeaderElectionConfiguration{
```
Comment (raghavendra-talur) [inline internal/controller/drcluster_mmode_test.go:100]: Rest of the changes look good to me. However, similar change as this is required in tests wherever we start a new manager. Search for `ctrl.NewManager`

Comment (ELENAGER) [reply internal/controller/drcluster_mmode_test.go:100]: @raghavendra-talur , yes, it is changed [here](https://github.com/ELENAGER/ramen/blob/7d893b743a8a8bdf2b1fd537658829efe0368e76/internal/controller/suite_test.go#L310), but not needed here[](https://github.com/ELENAGER/ramen/blob/7d893b743a8a8bdf2b1fd537658829efe0368e76/internal/controller/volsync/volsync_suite_test.go#L115)

**Review by ShyamsundarR — APPROVED** [review]

**Review by raghavendra-talur — APPROVED** [review]

**Pull request merged**